App.tsx

```diff
@@ -17,102 +19,66 @@ import {
   View,
 } from 'react-native';
 
-import {
-  Colors,
-  DebugInstructions,
-  Header,
-  LearnMoreLinks,
-  ReloadInstructions,
-} from 'react-native/Libraries/NewAppScreen';
 
-type SectionProps = PropsWithChildren<{
-  title: string;
-}>;
 
-function Section({children, title}: SectionProps): React.JSX.Element {
-  const isDarkMode = useColorScheme() === 'dark';
-  return (
-    <View style={styles.sectionContainer}>
-      <Text
-        style={[
-          styles.sectionTitle,
-          {
-            color: isDarkMode ? Colors.white : Colors.black,
-          },
-        ]}>
-        {title}
-      </Text>
-      <Text
-        style={[
-          styles.sectionDescription,
-          {
-            color: isDarkMode ? Colors.light : Colors.dark,
-          },
-        ]}>
-        {children}
-      </Text>
-    </View>
-  );
-}
+import Hea from './src/components/header';
+import Clear from './src/components/clear';
+import Section from './src/components/section';
+
+
 
 function App(): React.JSX.Element {
+
+  interface Element {
+    title: string;
+    description: string;
+    checked: boolean;
+  }
+
+  function selectCallback(isChecked: boolean, elem: Element) {
+    const actualizedTodo = todoS.map(e => {
+      if (e == elem) { e.checked = isChecked; }
+      return e;
+    })
+    setTodos(actualizedTodo);
+  }
+
+  function clearAllDone() {
+    const clearTodos = todoS.filter(obj => {
+      return obj.checked === false;
+    });
+    if (clearTodos.length !== todoS.length) {
+      setTodos(clearTodos);
+    }
+  }
+
   const isDarkMode = useColorScheme() === 'dark';
+  const [todoS, setTodos] = useState([
+    { title: 'prueba', description: 'hasdf', checked: false },
+    { title: 'prueba2', description: 'hasdf', checked: false },
+  ]);
```

Comment:
Since this array/objects is mocked data, what do you think of moving it to a separate variable outside of the component?

e.g.
Before the component:
```js
const mockedTodos = [
    { title: 'prueba', description: 'hasdf', checked: false },
    { title: 'prueba2', description: 'hasdf', checked: false },
  ]
```
(and even BETTER if you could type the Todo "model" as an interface, what do you think?)

Then in the component:
```suggestion
  const [todoS, setTodos] = useState(mockedTodo);
```

This way, it will be easier to switch between mocked data and the real integration with the backend data in the future.